tropohelper/services.py

```diff
@@ -26,6 +35,50 @@ def create_firehose(stack, name, bucket_arn, buffering_seconds=300, buffering_si
             RoleARN="arn:aws:iam::347225174248:role/firehose_delivery_role")
         ))
 
+def create_stream(stack, name, shard_count):
```

Comment:
Function name is ambiguous. What sort of stream? Also, missing doc block.

Reply:
renamed to create_kinesis_stream, doc also added

Reply:
Proof it can generate functional cloudformation json:
- ```docker run -it -p 4567-4578:4567-4578 -p 8080:8080 localstack/localstack```
- create stream:
```
aws --endpoint-url=http://localhost:4568 kinesis create-stream --stream-name sendgrid_stream --shard-count 5
```
- describe stream:
```
aws --endpoint-url=http://localhost:4568 kinesis describe-stream --stream-name sendgrid_stream
{
    "StreamDescription": {
        "KeyId": null, 
        "EncryptionType": "NONE", 
        "StreamStatus": "ACTIVE", 
        "StreamName": "sendgrid_stream", 
        "Shards": [
            {
                "ShardId": "shardId-000000000000", 
                "HashKeyRange": {
                    "EndingHashKey": "68056473384187692692674921486353642290", 
                    "StartingHashKey": "0"
                }, 
                "SequenceNumberRange": {
                    "StartingSequenceNumber": "49586844642679453023945388254643314865616851482739474434"
                }
            }, 
            {
                "ShardId": "shardId-000000000001", 
                "HashKeyRange": {
                    "EndingHashKey": "136112946768375385385349842972707284581", 
                    "StartingHashKey": "68056473384187692692674921486353642291"
                }, 
                "SequenceNumberRange": {
                    "StartingSequenceNumber": "49586844642701753769143918877784850583889499844245454866"
                }
            }, 
            {
                "ShardId": "shardId-000000000002", 
                "HashKeyRange": {
                    "EndingHashKey": "204169420152563078078024764459060926872", 
                    "StartingHashKey": "136112946768375385385349842972707284582"
                }, 
                "SequenceNumberRange": {
                    "StartingSequenceNumber": "49586844642724054514342449500926386302162148205751435298"
                }
            }, 
            {
                "ShardId": "shardId-000000000003", 
                "HashKeyRange": {
                    "EndingHashKey": "272225893536750770770699685945414569163", 
                    "StartingHashKey": "204169420152563078078024764459060926873"
                }, 
                "SequenceNumberRange": {
                    "StartingSequenceNumber": "49586844642746355259540980124067922020434796567257415730"
                }
            }, 
            {
                "ShardId": "shardId-000000000004", 
                "HashKeyRange": {
                    "EndingHashKey": "340282366920938463463374607431768211455", 
                    "StartingHashKey": "272225893536750770770699685945414569164"
                }, 
                "SequenceNumberRange": {
                    "StartingSequenceNumber": "49586844642768656004739510747209457738707444928763396162"
                }
            }
        ], 
        "StreamARN": "arn:aws:kinesis:us-east-1:000000000000:stream/sendgrid_stream", 
        "EnhancedMonitoring": [
            {
                "ShardLevelMetrics": []
            }
        ], 
        "StreamCreationTimestamp": 1532985025, 
        "RetentionPeriodHours": 24
    }
}
```
-- create redshift delivery stream:
```
aws --debug --endpoint-url=http://localhost:4573 firehose create-delivery-stream --delivery-stream-name sendgrid_delivery_stream_redshift --delivery-stream-type KinesisStreamAsSource     --kinesis-stream-source-configuration='{
        "KinesisStreamARN": "arn:aws:kinesis:us-east-1:000000000000:stream/sendgrid",
        "RoleARN": "random"
    }'     --redshift-destination-configuration='{
            "ClusterJDBCURL": "jdbc:redshift://localhost:4577",
            "CopyCommand": {
                "CopyOptions": "JSON",
                "DataTableName": "redshift_db_table_name"
            },
            "Password": "redshift_password",
            "RoleARN": "firehose_arn",
            "S3Configuration": {
                "BucketARN": "arn:aws:s3:::pp-dw-stg",
                "Prefix": "sendgrid",
                "BufferingHints": {
                    "SizeInMBs": 5,
                    "IntervalInSeconds": 60
                },
                "CompressionFormat": "GZIP",
                "EncryptionConfiguration": {
                    "KMSEncryptionConfig": {
                        "AWSKMSKeyARN": "arn:aws:kms:us-east-1:347225174248:key/232e45d1-6b15-4681-8de7-8629b0ed6b22"
                    }
                },
                "CloudWatchLoggingOptions": {
                    "Enabled": true,
                    "LogGroupName": "firehose-streams",
                    "LogStreamName": "name"
                },
                "RoleARN": "arn:aws:iam::347225174248:role/firehose_delivery_role"
            },
            "Username": "redshift"
    }'
2018-07-30 14:11:50,938 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/1.14.29 Python/2.7.10 Darwin/17.7.0 botocore/1.8.33
2018-07-30 14:11:50,938 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['--debug', '--endpoint-url=http://localhost:4573', 'firehose', 'create-delivery-stream', '--delivery-stream-name', 'sendgrid_delivery_stream_redshift', '--delivery-stream-type', 'KinesisStreamAsSource', '--kinesis-stream-source-configuration={\n        "KinesisStreamARN": "arn:aws:kinesis:us-east-1:000000000000:stream/sendgrid",\n        "RoleARN": "random"\n    }', '--redshift-destination-configuration={\n            "ClusterJDBCURL": "jdbc:redshift://localhost:4577",\n            "CopyCommand": {\n                "CopyOptions": "JSON",\n                "DataTableName": "redshift_db_table_name"\n            },\n            "Password": "redshift_password",\n            "RoleARN": "firehose_arn",\n            "S3Configuration": {\n                "BucketARN": "arn:aws:s3:::pp-dw-stg",\n                "Prefix": "sendgrid",\n                "BufferingHints": {\n                    "SizeInMBs": 5,\n                    "IntervalInSeconds": 60\n                },\n                "CompressionFormat": "GZIP",\n                "EncryptionConfiguration": {\n                    "KMSEncryptionConfig": {\n                        "AWSKMSKeyARN": "arn:aws:kms:us-east-1:347225174248:key/232e45d1-6b15-4681-8de7-8629b0ed6b22"\n                    }\n                },\n                "CloudWatchLoggingOptions": {\n                    "Enabled": true,\n                    "LogGroupName": "firehose-streams",\n                    "LogStreamName": "name"\n                },\n                "RoleARN": "arn:aws:iam::347225174248:role/firehose_delivery_role"\n            },\n            "Username": "redshift"\n    }']
2018-07-30 14:11:50,938 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function add_scalar_parsers at 0x10b666230>
2018-07-30 14:11:50,938 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,938 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function inject_assume_role_provider_cache at 0x10b04e140>
2018-07-30 14:11:50,938 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,938 - MainThread - botocore.session - DEBUG - Loading variable credentials_file from defaults.
2018-07-30 14:11:50,938 - MainThread - botocore.session - DEBUG - Loading variable config_file from defaults.
2018-07-30 14:11:50,938 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,938 - MainThread - botocore.session - DEBUG - Loading variable metadata_service_timeout from defaults.
2018-07-30 14:11:50,938 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,938 - MainThread - botocore.session - DEBUG - Loading variable metadata_service_num_attempts from defaults.
2018-07-30 14:11:50,939 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,939 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function attach_history_handler at 0x10b267ed8>
2018-07-30 14:11:50,939 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,939 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,939 - MainThread - botocore.session - DEBUG - Loading variable api_versions from defaults.
2018-07-30 14:11:50,949 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /Users/ara.abrahamian/Library/Python/2.7/lib/python/site-packages/botocore/data/firehose/2015-08-04/service-2.json
2018-07-30 14:11:50,956 - MainThread - botocore.hooks - DEBUG - Event service-data-loaded.firehose: calling handler <function register_retries_for_service at 0x10ac3e2a8>
2018-07-30 14:11:50,957 - MainThread - botocore.handlers - DEBUG - Registering retry handlers for service: firehose
2018-07-30 14:11:50,957 - MainThread - botocore.hooks - DEBUG - Event building-command-table.firehose: calling handler <function add_waiters at 0x10b66c320>
2018-07-30 14:11:50,964 - MainThread - awscli.clidriver - DEBUG - OrderedDict([(u'delivery-stream-name', <awscli.arguments.CLIArgument object at 0x10b8492d0>), (u'delivery-stream-type', <awscli.arguments.CLIArgument object at 0x10b849310>), (u'kinesis-stream-source-configuration', <awscli.arguments.CLIArgument object at 0x10b849350>), (u's3-destination-configuration', <awscli.arguments.CLIArgument object at 0x10b849390>), (u'extended-s3-destination-configuration', <awscli.arguments.CLIArgument object at 0x10b849410>), (u'redshift-destination-configuration', <awscli.arguments.CLIArgument object at 0x10b849450>), (u'elasticsearch-destination-configuration', <awscli.arguments.CLIArgument object at 0x10b849490>), (u'splunk-destination-configuration', <awscli.arguments.CLIArgument object at 0x10b8494d0>)])
2018-07-30 14:11:50,964 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.firehose.create-delivery-stream: calling handler <function add_streaming_output_arg at 0x10b666488>
2018-07-30 14:11:50,964 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.firehose.create-delivery-stream: calling handler <function add_cli_input_json at 0x10b04ec80>
2018-07-30 14:11:50,965 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.firehose.create-delivery-stream: calling handler <function unify_paging_params at 0x10b2f1758>
2018-07-30 14:11:50,971 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /Users/ara.abrahamian/Library/Python/2.7/lib/python/site-packages/botocore/data/firehose/2015-08-04/paginators-1.json
2018-07-30 14:11:50,971 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.firehose.create-delivery-stream: calling handler <function add_generate_skeleton at 0x10b2ddb18>
2018-07-30 14:11:50,971 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.firehose.create-delivery-stream: calling handler <bound method CliInputJSONArgument.override_required_args of <awscli.customizations.cliinputjson.CliInputJSONArgument object at 0x10b849510>>
2018-07-30 14:11:50,971 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.firehose.create-delivery-stream: calling handler <bound method GenerateCliSkeletonArgument.override_required_args of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x10b8496d0>>
2018-07-30 14:11:50,972 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.firehose.create-delivery-stream.delivery-stream-name: calling handler <function uri_param at 0x10aee98c0>
2018-07-30 14:11:50,972 - MainThread - botocore.hooks - DEBUG - Event process-cli-arg.firehose.create-delivery-stream: calling handler <awscli.argprocess.ParamShorthandParser object at 0x10b67a310>
2018-07-30 14:11:50,972 - MainThread - awscli.arguments - DEBUG - Unpacked value of u'sendgrid_delivery_stream_redshift' for parameter "delivery_stream_name": u'sendgrid_delivery_stream_redshift'
2018-07-30 14:11:50,972 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.firehose.create-delivery-stream.delivery-stream-type: calling handler <function uri_param at 0x10aee98c0>
2018-07-30 14:11:50,972 - MainThread - botocore.hooks - DEBUG - Event process-cli-arg.firehose.create-delivery-stream: calling handler <awscli.argprocess.ParamShorthandParser object at 0x10b67a310>
2018-07-30 14:11:50,972 - MainThread - awscli.arguments - DEBUG - Unpacked value of u'KinesisStreamAsSource' for parameter "delivery_stream_type": u'KinesisStreamAsSource'
2018-07-30 14:11:50,972 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.firehose.create-delivery-stream.kinesis-stream-source-configuration: calling handler <function uri_param at 0x10aee98c0>
2018-07-30 14:11:50,972 - MainThread - botocore.hooks - DEBUG - Event process-cli-arg.firehose.create-delivery-stream: calling handler <awscli.argprocess.ParamShorthandParser object at 0x10b67a310>
2018-07-30 14:11:50,972 - MainThread - awscli.argprocess - DEBUG - Param kinesis_stream_source_configuration looks like JSON, not considered for param shorthand.
2018-07-30 14:11:50,973 - MainThread - awscli.arguments - DEBUG - Unpacked value of u'{\n        "KinesisStreamARN": "arn:aws:kinesis:us-east-1:000000000000:stream/sendgrid",\n        "RoleARN": "random"\n    }' for parameter "kinesis_stream_source_configuration": OrderedDict([(u'KinesisStreamARN', u'arn:aws:kinesis:us-east-1:000000000000:stream/sendgrid'), (u'RoleARN', u'random')])
2018-07-30 14:11:50,973 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.firehose.create-delivery-stream.s3-destination-configuration: calling handler <function uri_param at 0x10aee98c0>
2018-07-30 14:11:50,973 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.firehose.create-delivery-stream.extended-s3-destination-configuration: calling handler <function uri_param at 0x10aee98c0>
2018-07-30 14:11:50,973 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.firehose.create-delivery-stream.redshift-destination-configuration: calling handler <function uri_param at 0x10aee98c0>
2018-07-30 14:11:50,973 - MainThread - botocore.hooks - DEBUG - Event process-cli-arg.firehose.create-delivery-stream: calling handler <awscli.argprocess.ParamShorthandParser object at 0x10b67a310>
2018-07-30 14:11:50,973 - MainThread - awscli.argprocess - DEBUG - Param redshift_destination_configuration looks like JSON, not considered for param shorthand.
2018-07-30 14:11:50,973 - MainThread - awscli.arguments - DEBUG - Unpacked value of u'{\n            "ClusterJDBCURL": "jdbc:redshift://localhost:4577",\n            "CopyCommand": {\n                "CopyOptions": "JSON",\n                "DataTableName": "redshift_db_table_name"\n            },\n            "Password": "redshift_password",\n            "RoleARN": "firehose_arn",\n            "S3Configuration": {\n                "BucketARN": "arn:aws:s3:::pp-dw-stg",\n                "Prefix": "sendgrid",\n                "BufferingHints": {\n                    "SizeInMBs": 5,\n                    "IntervalInSeconds": 60\n                },\n                "CompressionFormat": "GZIP",\n                "EncryptionConfiguration": {\n                    "KMSEncryptionConfig": {\n                        "AWSKMSKeyARN": "arn:aws:kms:us-east-1:347225174248:key/232e45d1-6b15-4681-8de7-8629b0ed6b22"\n                    }\n                },\n                "CloudWatchLoggingOptions": {\n                    "Enabled": true,\n                    "LogGroupName": "firehose-streams",\n                    "LogStreamName": "name"\n                },\n                "RoleARN": "arn:aws:iam::347225174248:role/firehose_delivery_role"\n            },\n            "Username": "redshift"\n    }' for parameter "redshift_destination_configuration": OrderedDict([(u'ClusterJDBCURL', u'jdbc:redshift://localhost:4577'), (u'CopyCommand', OrderedDict([(u'CopyOptions', u'JSON'), (u'DataTableName', u'redshift_db_table_name')])), (u'Password', u'redshift_password'), (u'RoleARN', u'firehose_arn'), (u'S3Configuration', OrderedDict([(u'BucketARN', u'arn:aws:s3:::pp-dw-stg'), (u'Prefix', u'sendgrid'), (u'BufferingHints', OrderedDict([(u'SizeInMBs', 5), (u'IntervalInSeconds', 60)])), (u'CompressionFormat', u'GZIP'), (u'EncryptionConfiguration', OrderedDict([(u'KMSEncryptionConfig', OrderedDict([(u'AWSKMSKeyARN', u'arn:aws:kms:us-east-1:347225174248:key/232e45d1-6b15-4681-8de7-8629b0ed6b22')]))])), (u'CloudWatchLoggingOptions', OrderedDict([(u'Enabled', True), (u'LogGroupName', u'firehose-streams'), (u'LogStreamName', u'name')])), (u'RoleARN', u'arn:aws:iam::347225174248:role/firehose_delivery_role')])), (u'Username', u'redshift')])
2018-07-30 14:11:50,974 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.firehose.create-delivery-stream.elasticsearch-destination-configuration: calling handler <function uri_param at 0x10aee98c0>
2018-07-30 14:11:50,974 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.firehose.create-delivery-stream.splunk-destination-configuration: calling handler <function uri_param at 0x10aee98c0>
2018-07-30 14:11:50,974 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.firehose.create-delivery-stream.cli-input-json: calling handler <function uri_param at 0x10aee98c0>
2018-07-30 14:11:50,974 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.firehose.create-delivery-stream.generate-cli-skeleton: calling handler <function uri_param at 0x10aee98c0>
2018-07-30 14:11:50,974 - MainThread - botocore.hooks - DEBUG - Event calling-command.firehose.create-delivery-stream: calling handler <bound method CliInputJSONArgument.add_to_call_parameters of <awscli.customizations.cliinputjson.CliInputJSONArgument object at 0x10b849510>>
2018-07-30 14:11:50,974 - MainThread - botocore.hooks - DEBUG - Event calling-command.firehose.create-delivery-stream: calling handler <bound method GenerateCliSkeletonArgument.generate_json_skeleton of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x10b8496d0>>
2018-07-30 14:11:50,974 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,974 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,974 - MainThread - botocore.session - DEBUG - Loading variable region from config file with value 'us-east-1'.
2018-07-30 14:11:50,974 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,974 - MainThread - botocore.session - DEBUG - Loading variable ca_bundle from defaults.
2018-07-30 14:11:50,974 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,974 - MainThread - botocore.session - DEBUG - Loading variable api_versions from defaults.
2018-07-30 14:11:50,974 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: env
2018-07-30 14:11:50,974 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: assume-role
2018-07-30 14:11:50,974 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: shared-credentials-file
2018-07-30 14:11:50,975 - MainThread - botocore.credentials - INFO - Found credentials in shared credentials file: ~/.aws/credentials
2018-07-30 14:11:50,975 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /Users/ara.abrahamian/Library/Python/2.7/lib/python/site-packages/botocore/data/endpoints.json
2018-07-30 14:11:50,991 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:50,991 - MainThread - botocore.hooks - DEBUG - Event choose-service-name: calling handler <function handle_service_name_alias at 0x10ac3ba28>
2018-07-30 14:11:50,991 - MainThread - botocore.hooks - DEBUG - Event creating-client-class.firehose: calling handler <function add_generate_presigned_url at 0x10ac07f50>
2018-07-30 14:11:50,992 - MainThread - botocore.args - DEBUG - The s3 config key is not a dictionary type, ignoring its value of: None
2018-07-30 14:11:50,999 - MainThread - botocore.endpoint - DEBUG - Setting firehose timeout as (60, 60)
2018-07-30 14:11:51,000 - MainThread - botocore.client - DEBUG - Registering retry handlers for service: firehose
2018-07-30 14:11:51,000 - MainThread - botocore.hooks - DEBUG - Event before-parameter-build.firehose.CreateDeliveryStream: calling handler <function generate_idempotent_uuid at 0x10ac3bc80>
2018-07-30 14:11:51,002 - MainThread - botocore.endpoint - DEBUG - Making request for OperationModel(name=CreateDeliveryStream) (verify_ssl=True) with params: {'body': '{"KinesisStreamSourceConfiguration": {"RoleARN": "random", "KinesisStreamARN": "arn:aws:kinesis:us-east-1:000000000000:stream/sendgrid"}, "DeliveryStreamType": "KinesisStreamAsSource", "DeliveryStreamName": "sendgrid_delivery_stream_redshift", "RedshiftDestinationConfiguration": {"Username": "redshift", "S3Configuration": {"BucketARN": "arn:aws:s3:::pp-dw-stg", "Prefix": "sendgrid", "BufferingHints": {"IntervalInSeconds": 60, "SizeInMBs": 5}, "EncryptionConfiguration": {"KMSEncryptionConfig": {"AWSKMSKeyARN": "arn:aws:kms:us-east-1:347225174248:key/232e45d1-6b15-4681-8de7-8629b0ed6b22"}}, "CompressionFormat": "GZIP", "CloudWatchLoggingOptions": {"Enabled": true, "LogStreamName": "name", "LogGroupName": "firehose-streams"}, "RoleARN": "arn:aws:iam::347225174248:role/firehose_delivery_role"}, "RoleARN": "firehose_arn", "ClusterJDBCURL": "jdbc:redshift://localhost:4577", "CopyCommand": {"DataTableName": "redshift_db_table_name", "CopyOptions": "JSON"}, "Password": "redshift_password"}}', 'url': u'http://localhost:4573/', 'headers': {'User-Agent': 'aws-cli/1.14.29 Python/2.7.10 Darwin/17.7.0 botocore/1.8.33', 'Content-Type': u'application/x-amz-json-1.1', 'X-Amz-Target': u'Firehose_20150804.CreateDeliveryStream'}, 'context': {'auth_type': None, 'client_region': 'us-east-1', 'has_streaming_input': False, 'client_config': <botocore.config.Config object at 0x10b858550>}, 'query_string': '', 'url_path': '/', 'method': u'POST'}
2018-07-30 14:11:51,002 - MainThread - botocore.hooks - DEBUG - Event request-created.firehose.CreateDeliveryStream: calling handler <bound method RequestSigner.handler of <botocore.signers.RequestSigner object at 0x10b858510>>
2018-07-30 14:11:51,003 - MainThread - botocore.hooks - DEBUG - Event choose-signer.firehose.CreateDeliveryStream: calling handler <function set_operation_specific_signer at 0x10ac3bb90>
2018-07-30 14:11:51,003 - MainThread - botocore.auth - DEBUG - Calculating signature using v4 auth.
2018-07-30 14:11:51,003 - MainThread - botocore.auth - DEBUG - CanonicalRequest:
POST
/

content-type:application/x-amz-json-1.1
host:localhost:4573
x-amz-date:20180730T211151Z
x-amz-target:Firehose_20150804.CreateDeliveryStream

content-type;host;x-amz-date;x-amz-target
3417d8da29181de340ea950c629cb83cb7deaf7b4221a279c7d344be8af2f745
2018-07-30 14:11:51,003 - MainThread - botocore.auth - DEBUG - StringToSign:
AWS4-HMAC-SHA256
20180730T211151Z
20180730/us-east-1/firehose/aws4_request
648f8a3a13483577e7627aa5788b7b37f1c398113f8af493213f8c4bb5956193
2018-07-30 14:11:51,003 - MainThread - botocore.auth - DEBUG - Signature:
a89be5e6296756386a666fe61a9ad61de0929901a565e7589776afd8a7ef84ea
2018-07-30 14:11:51,005 - MainThread - botocore.endpoint - DEBUG - Sending http request: <PreparedRequest [POST]>
2018-07-30 14:11:51,006 - MainThread - botocore.vendored.requests.packages.urllib3.connectionpool - INFO - Starting new HTTP connection (1): localhost
2018-07-30 14:11:51,014 - MainThread - botocore.vendored.requests.packages.urllib3.connectionpool - DEBUG - "POST / HTTP/1.1" 200 334
2018-07-30 14:11:51,015 - MainThread - botocore.parsers - DEBUG - Response headers: {'date': 'Mon, 30 Jul 2018 21:11:51 GMT', 'content-length': '334', 'content-type': 'application/json', 'access-control-allow-origin': '*', 'server': 'Werkzeug/0.14.1 Python/2.7.13'}
2018-07-30 14:11:51,015 - MainThread - botocore.parsers - DEBUG - Response body:
{
  "CreateTimestamp": 1532985111.017871, 
  "DeliveryStreamARN": "arn:aws:firehose:us-east-1:000000000000:deliverystream/sendgrid_delivery_stream_redshift", 
  "DeliveryStreamName": "sendgrid_delivery_stream_redshift", 
  "DeliveryStreamStatus": "ACTIVE", 
  "Destinations": [], 
  "HasMoreDestinations": false, 
  "VersionId": "1"
}
2018-07-30 14:11:51,015 - MainThread - botocore.hooks - DEBUG - Event needs-retry.firehose.CreateDeliveryStream: calling handler <botocore.retryhandler.RetryHandler object at 0x10b77f150>
2018-07-30 14:11:51,015 - MainThread - botocore.retryhandler - DEBUG - No retry needed.
2018-07-30 14:11:51,015 - MainThread - botocore.session - DEBUG - Loading variable profile from defaults.
2018-07-30 14:11:51,015 - MainThread - botocore.session - DEBUG - Loading variable output from defaults.
{
    "DeliveryStreamARN": "arn:aws:firehose:us-east-1:000000000000:deliverystream/sendgrid_delivery_stream_redshift"
}
```

Reply:
unit test added